### PR TITLE
add documentation for the docker container generations scripts

### PR DESCRIPTION
**Pull request by bam241**
this provide some help and instructions on how to use the Dockerfile and associated python script in order to generate and push the PyNE CI docker container.

### CODE REVIEW

File: docker/readme.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    
+    - "--py_version=X" with "X" = 2 or 3, specify the python version
+    - "--moab" install MOAB in the container
+    - "--dagmc" install DAGMC in the container
+    - "--pymoab" install pyMOAB in the container (along with MOAB)
+    - "--all/-a/-all" install all the optional dependencies (MOAB/pyMOAB/DAGMC)
+    - "--deps" don't install PyNE only install the selected optional PyNE dependencies (along with the required one)
+    - "--push" push the docker container to DockerHub after building it
+      (requires right to push on the PyNE DockerHub account)
+    - "-name=" manually set the docker container name (not recommended with used
```
Comment (zxkjack123) [inline docker/readme.md]:
```suggestion
    - "--name=" manually set the docker container name (NOT recommended with used
```

File: docker/readme.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      with the push option)
+
+
+The Python script will check consistency in the required dependency to ensure
+working build: pyMOAB requires MOAB to be install (MOAB will be install if
+pyMOAB flag is provided, even if MOAB flag is not), DAGMC requires pyMOAB (and
+MOAB).
+
+The Python script will also also form the container name accordingly to some
```
Comment (zxkjack123) [inline docker/readme.md]:
```suggestion
The Python script will also name container according to some
```

File: docker/readme.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+convention allowing the automatic CI to use the pushed docker image, One
+**STRONGLY** advices against manually set the container name when updating the CI
+docker containers in the PyNE DockerHuB. 
+
+
+One recommends to maintainers to use the python script to generate and push the docker
```
Comment (zxkjack123) [inline docker/readme.md]:
```suggestion
We recommend maintainers use the python script to generate and push the docker
```

File: docker/readme.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+Docker Script Documentation
+===========================
+
+The files in this folder are intended to be mainly used by PyNE maintainers,
+they provide an "easy" mechanism to build and push the different containers
+required for the multiple configurations test by the PyNE CI.
+
+This folder contains:
+
+- A custom Dockerfile: allowing to build docker containers with a custom set of
```
Comment (zxkjack123) [inline docker/readme.md]:
```suggestion
- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing to build docker containers with a custom set of
```

File: docker/readme.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  can be specified using the "--build-args" flags. The different flags are:
+    
+    - "build_moab=YES" add MOAB in the docker container
+    - "build_pymoab=YES" install MOAB with pymoab
+    - "build_dagmc=YES" add DAGMC 
+    - "build_pyne=NO" only add the PyNE dependencies to the container
+    - "py_version=X.Y" specify the python version to install (option are 2.7 or
+      3.6)
+
+- A python script is also present to simplify the usage of the docker file, it
```
Comment (zxkjack123) [inline docker/readme.md]:
```suggestion
- A python script, "make_pyne_docker_image.py", is also present to simplify the usage of the docker file, it
```

**Review by zxkjack123 — COMMENTED** [review]
Some minor suggestions as address in the suggested changes. By the way, 'rst' format is widely used in PyNE repo, I think it's better to name this file as "readme.rst".

**Comment by bam241** [comment]
Thx for the suggestion @zxkjack123.

I would feel more comfortable if a native speaker could also review it :)

**Comment by bam241** [comment]
cc @pshriwise @gonuke

File: docker/readme.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+Docker Script Documentation
+===========================
+
+The files in this folder are intended to be mainly used by PyNE maintainers,
+they provide an "easy" mechanism to build and push the different containers
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
they provide an "easy" mechanism to build and push the various containers
```

File: docker/readme.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+required for the multiple configurations test by the PyNE CI.
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
required for the multiple configurations tested by the PyNE CI.
```

File: docker/readme.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+This folder contains:
+
+- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing to build docker containers with a custom set of
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing the user 
to build docker containers with a custom set of
```

File: docker/readme.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  dependencies required by the CI, the configuration of the Docker containers
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
  dependencies required by the CI. The configuration of the Docker containers
```

File: docker/readme.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  can be specified using the "--build-args" flags. The different flags are:
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
  can be specified using the `--build_***` flags. The available flags are:
```

File: docker/readme.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    
+    - "build_moab=YES" add MOAB in the docker container
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `build_moab=YES` add MOAB in the docker container
```

File: docker/readme.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    - "build_pymoab=YES" install MOAB with pymoab
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `build_pymoab=YES` install MOAB with pymoab
```

File: docker/readme.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    - "build_dagmc=YES" add DAGMC
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `build_dagmc=YES` add DAGMC 
```

File: docker/readme.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+This folder contains:
+
+- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing to build docker containers with a custom set of
+  dependencies required by the CI, the configuration of the Docker containers
+  can be specified using the "--build-args" flags. The different flags are:
+    
+    - "build_moab=YES" add MOAB in the docker container
+    - "build_pymoab=YES" install MOAB with pymoab
+    - "build_dagmc=YES" add DAGMC 
+    - "build_pyne=NO" only add the PyNE dependencies to the container
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `build_pyne=NO` do not build PyNE, only add the PyNE dependencies to the container
```

File: docker/readme.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+    - "py_version=X.Y" specify the python version to install (option are 2.7 or
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `py_version=X.Y` specify the python version to install (options are 2.7 or
```

File: docker/readme.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+      3.6)
+
+- A python script, "make_pyne_docker_image.py", is also present to simplify the usage of the docker file, it
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
- A python script, `make_pyne_docker_image.py`, is also present to simplify the usage of the docker file. It
```

File: docker/readme.md [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+  will allows maintainers to build, name and push the docker container into
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
 allows maintainers to build, name, and push the docker container into
```

File: docker/readme.md [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+  DockerHub. The different option are:
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
  DockerHub. The available options are:
```

File: docker/readme.md [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    - "--pymoab" install pyMOAB in the container (along with MOAB)
+    - "--all/-a/-all" install all the optional dependencies (MOAB/pyMOAB/DAGMC)
+    - "--deps" don't install PyNE only install the selected optional PyNE dependencies (along with the required one)
+    - "--push" push the docker container to DockerHub after building it
+      (requires right to push on the PyNE DockerHub account)
+    - "--name=" manually set the docker container name (NOT recommended with used
+      with the push option)
+
+
+The Python script will check consistency in the required dependency to ensure
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
The Python script will check for consistency among the required dependencies to ensure
```

File: docker/readme.md [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+working build: pyMOAB requires MOAB to be install (MOAB will be install if
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
working build. For example, pyMOAB requires MOAB to be install (MOAB will be installed if
```

File: docker/readme.md [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+pyMOAB flag is provided, even if MOAB flag is not), DAGMC requires pyMOAB (and
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
the pyMOAB flag is provided, even if the MOAB flag is not), DAGMC requires pyMOAB (and
```

File: docker/readme.md [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+MOAB).
+
+The Python script will also name container according to some
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
The Python script will also name the container according to the
```

File: docker/readme.md [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+convention allowing the automatic CI to use the pushed docker image, One
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
convention, allowing the automatic CI to use the pushed docker image. We
```

File: docker/readme.md [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+**STRONGLY** advices against manually set the container name when updating the CI
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
**STRONGLY** advise against manually setting the container name when updating the CI
```

File: docker/readme.md [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+docker containers in the PyNE DockerHuB. 
+
+
+We recommend maintainers use the python script to generate and push the docker
+container in order to form container name correctly, and include the proper
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
container in order to form container name correctly and to include the proper
```

File: docker/readme.md [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+    - "build_dagmc=YES" add DAGMC 
+    - "build_pyne=NO" only add the PyNE dependencies to the container
+    - "py_version=X.Y" specify the python version to install (option are 2.7 or
+      3.6)
+
+- A python script, "make_pyne_docker_image.py", is also present to simplify the usage of the docker file, it
+  will allows maintainers to build, name and push the docker container into
+  DockerHub. The different option are:
+    
+    - "--py_version=X" with "X" = 2 or 3, specify the python version
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--py_version=X` with "X" = major python version (e.g. 2 or 3)
```

File: docker/readme.md [hunk 27]
```diff
@@ -0,0 +1,50 @@
...
+    - "--moab" install MOAB in the container
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--moab` install MOAB in the container
```

File: docker/readme.md [hunk 28]
```diff
@@ -0,0 +1,50 @@
...
+    - "--dagmc" install DAGMC in the container
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--dagmc` install DAGMC in the container
```

File: docker/readme.md [hunk 29]
```diff
@@ -0,0 +1,50 @@
...
+    - "--pymoab" install pyMOAB in the container (along with MOAB)
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--pymoab` install pyMOAB in the container (along with MOAB)
```

File: docker/readme.md [hunk 30]
```diff
@@ -0,0 +1,50 @@
...
+    - "--all/-a/-all" install all the optional dependencies (MOAB/pyMOAB/DAGMC)
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--all/-a/-all` install all the optional dependencies (MOAB/pyMOAB/DAGMC)
```

File: docker/readme.md [hunk 31]
```diff
@@ -0,0 +1,50 @@
...
+    - "--deps" don't install PyNE only install the selected optional PyNE dependencies (along with the required one)
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--deps` don't install PyNE, only install the dependencies (both required and selected optional dependencies)
```

File: docker/readme.md [hunk 32]
```diff
@@ -0,0 +1,50 @@
...
+    - "--push" push the docker container to DockerHub after building it
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--push` push the docker container to DockerHub after building it
```

File: docker/readme.md [hunk 33]
```diff
@@ -0,0 +1,50 @@
...
+      (requires right to push on the PyNE DockerHub account)
+    - "--name=" manually set the docker container name (NOT recommended with used
```
Comment (katyhuff) [inline docker/readme.md]:
```suggestion
    - `--name=` manually set the docker container name (NOT recommended to be used
```

**Review by katyhuff — CHANGES_REQUESTED** [review]

File: docker/readme.md (hunk 10 above)
Comment (bam241) [reply docker/readme.md]: the correct usage in docker is actually `--build_args build_moab=YES`.

maybe `--build_args build_***=YES` is the good way to write it ?
or `--build_args *********` ?

Comment (katyhuff) [reply docker/readme.md]: ah -- gotcha... that was not clear. I get it now.

If that's the case, then this place should, indeed, say [`--build-arg <flag>`](https://github.com/pyne/pyne/blob/3374b304124dd345b91c99ed3b8aa8b32fc7d85e/docker/make_pyne_docker_image.py#L66) .

File: docker/readme.md [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+    
+    - `build_moab=YES` add MOAB in the docker container
+    - `build_pymoab=YES` install MOAB with pymoab
+    - `build_dagmc=YES` add DAGMC 
+    - `build_pyne=NO` do not build PyNE, only add the PyNE dependencies to the container
+    - `py_version=X.Y` specify the python version to install (options are 2.7 or
+      3.6)
+Note, if using multiple build arguments `--build-args` has to be repeted, i.e. :
+to request DAGMC and only dependencies with Python 2.7, the docker build arguments will be: 
+`--build-args build_dagmc=YES --build-args build_pyne=NO --build-args=py_version=2.7`
```
Comment (bam241) [inline docker/readme.md]: Does it look better @katyhuff ? 
(I also added an example...)

(thx for the review and the English corrections!)

File: docker/readme.md [hunk 35]
```diff
@@ -0,0 +1,54 @@
...
+  can be specified using the `--build-args <flag>` argument. The different flags are:
+    
+    - `build_moab=YES` add MOAB in the docker container
+    - `build_pymoab=YES` install MOAB with pymoab
+    - `build_dagmc=YES` add DAGMC 
+    - `build_pyne=NO` do not build PyNE, only add the PyNE dependencies to the container
+    - `py_version=X.Y` specify the python version to install (options are 2.7 or
+      3.6)
+Note, if using multiple build arguments `--build-args` has to be repeted, i.e. :
+to request DAGM, dependenciesonly with Python 2.7, the docker build arguments will be:
```
Comment (zxkjack123) [inline docker/readme.md]:
```suggestion
to request DAGMC dependencies only with Python 2.7, the docker build arguments will be: 
```

File: docker/readme.md [hunk 36]
```diff
@@ -0,0 +1,54 @@
...
+    
+    - `build_moab=YES` add MOAB in the docker container
+    - `build_pymoab=YES` install MOAB with pymoab
+    - `build_dagmc=YES` add DAGMC 
+    - `build_pyne=NO` do not build PyNE, only add the PyNE dependencies to the container
+    - `py_version=X.Y` specify the python version to install (options are 2.7 or
+      3.6)
+Note, if using multiple build arguments `--build-args` has to be repeted, i.e. :
+to request DAGMC dependencies only with Python 2.7, the docker build arguments will be: 
+`--build-args build_dagmc=YES --build-args build_pyne=NO --build-args py_version=2.7`
```
Comment (bam241) [inline docker/readme.md:25]: Does it look better ?
(I also added an example...)

(thx for the review and the English corrections!)

File: docker/readme.md [hunk 37]
```diff
@@ -0,0 +1,54 @@
...
+required for the multiple configurations tested by the PyNE CI.
+
+This folder contains:
+
+- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing the user 
+to build docker containers with a custom set of
+  dependencies required by the CI. The configuration of the Docker containers
+  can be specified using the `--build-args <flag>` argument. The different flags are:
+    
+    - `build_moab=YES` add MOAB in the docker container
```
Comment (gonuke) [inline docker/readme.md]: Are these the default values?  Presumably you can say either `YES` or `NO` for each option?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @bam241.

I think it's important to be clear about what the default values are if no `build_args` are provided.

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
Thanks @bam241  - I think you’ve addressed all of @katyhuff  requests.

**Pull request merged**